### PR TITLE
Implement clear internship command

### LINKED ISSUE #116: As a CS student, I can delete all internship applications to a company I am no longer interested in...
so I can declutter my internship application list.

### LINKED ISSUE #130: As a CS student, I can clear the internship list of a specific company

### PR DESCRIPTION
**Pull request by rngzhi**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/clearcommands/ClearCommand.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    public static final String COMMAND_WORD = "clear";
+    public static final String MESSAGE_SUCCESS = "Success message for the clear command";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Clears person list, company list, "
+            + "internship list of a specific company"
+            + "\n"
+            + "Parameters: \n"
+            + "c - Clears all companies\n"
+            + "p - Clears all persons \n"
+            + "i - Clears all internships\n"
+            + "Example: " + COMMAND_WORD + " p";
```
Comment (Tim-Siu) [inline src/main/java/seedu/address/logic/commands/clearcommands/ClearCommand.java]: The message for clearing internships of a company seem weird?

File: src/main/java/seedu/address/logic/parser/ClearCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        String[] typeIndex = trimmedArgs.split("\\s+");
+        if (typeIndex.length != 2) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ClearCommand.MESSAGE_USAGE));
+        }
+        String type = typeIndex[0];
+        logger.info("type: " + type);
+
+        // Used to check if type is either c or p.
```
Comment (Tim-Siu) [inline src/main/java/seedu/address/logic/parser/ClearCommandParser.java]: It seems to be a typo

File: src/test/java/seedu/address/logic/commands/clearcommands/ClearInternshipCommandTest.java
Comment (Tim-Siu) [inline src/test/java/seedu/address/logic/commands/clearcommands/ClearInternshipCommandTest.java:1]: It seems like there is a lack of actual integration tests. Hopefully, the system can work.

File: src/main/java/seedu/address/logic/commands/clearcommands/ClearCommand.java (hunk 1 above)
Comment (rngzhi) [reply src/main/java/seedu/address/logic/commands/clearcommands/ClearCommand.java]: Thanks for pointing out, I have rephrased the message in the new commit.

File: src/main/java/seedu/address/logic/parser/ClearCommandParser.java (hunk 2 above)
Comment (rngzhi) [reply src/main/java/seedu/address/logic/parser/ClearCommandParser.java]: Also fixed in the new commit.

**Review by Tim-Siu — APPROVED** [review]

**Comment by Tim-Siu** [comment]
Thank you Rongzhi for your contribution. The OOP design around DeleteCommand will facilitate further extention.

**Pull request merged**